docs/reference/upgrade.md

```diff
@@ -11,11 +11,19 @@ redirect_from: ../deploying-aws/upgrade.html
 # Upgrading lakeFS
 {: .no_toc }
 Upgrading lakeFS from a previous version usually just requires re-deploying with the latest image (or downloading the latest version if you're using the binary).
-In some cases, the database may require a migration - check whether the [release](https://github.com/treeverse/lakeFS/releases) you're upgrading to requires that.
+If you're upgrading, check whether the [release](https://github.com/treeverse/lakeFS/releases) requires a migration.
 
 
 ## When DB migrations are required
 
+### lakeFS <TBD.X.X> or greater (KV Migration)
+
+Starting with version <TBD.X.X>, lakeFS has transitioned from using a PostgreSQL based database implementation to a Key-Value datastore interface supporting
+multiple database implementations. More information can be found [here](link-to-lakefs-on-kv-documentation).  
+Users upgrading from a previous version of lakeFS must pass through the KV migration version (<TBD.X.X>) before upgrading to newer versions of lakeFS.
+The migration process is identical to the process described [here](#lakefs-0300-or-greater) and ensures a clean transition from the old metadata database model to the new one.
```

Comment:
I rather we mention the explicit steps to be made instead of referring to the old (although similar) migration.

Reply:
https://github.com/treeverse/lakeFS/pull/3878#issuecomment-1210531699

Reply:
Done